## <span style="color: #008000;">Introduction: FitBit Tracker Data Analysis for Bella Beats&nbsp;</span> 

### <span style="color: #008000;">Notes about the data:</span>

<span style="color: #008000;">The data being used in this case study was retrieved from Kaggle and had been uploaded by a Kaggler named Möbius.&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">The data source is licensed under a public domain so copyright and IP concerns should not be an issue. For more information&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">about the data source and this public license please visit the following link: </span> [https://www.kaggle.com/datasets/arashnic/fitbit](https://www.kaggle.com/datasets/arashnic/fitbit)

### <span style="color: #008000;">Background on the data:</span>

<span style="color: #008000;">The datasets available in this datasource contains health-related information retrieved from roughly 30 'eligible' FitBit users who had&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">consented to sharing their FitBit tracking data to the public. Upon previewing this source we notice that there were 18 different tables&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">containing data on different categories and even subsets of health-related information. The main datasets we're concerned with for the&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">purposes of this project include the data involving: (1) Activity, which is found in the Daily_Activity table; (2) sleep, which is found&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">in the Daily_Sleep table; and finally (3) Weight, which is found in the WeightLog table.</span>

In [1]:
-- Previewing the Daily_Activity dataset

SELECT 
    TOP 5 *
FROM
    CaseStudy1.dbo.Daily_Activity
ORDER BY 
    ActivityDate ASC

-- Previewing the Daily_Sleep dataset

SELECT 
    TOP 5 *
FROM CaseStudy1.dbo.Daily_Sleep
ORDER BY
    SleepDay ASC

-- Previewing the WeightLog dataset

SELECT TOP 5 *
FROM 
    CaseStudy1.dbo.WeightLog
ORDER BY 
    Date ASC


(5 rows affected)

(5 rows affected)

(5 rows affected)

Total execution time: 00:00:00.153

Id ActivityDate TotalSteps TotalDistance TrackerDistance LoggedActivitiesDistance VeryActiveDistance ModeratelyActiveDistance LightActiveDistance SedentaryActiveDistance VeryActiveMinutes FairlyActiveMinutes LightlyActiveMinutes SedentaryMinutes Calories 1503960366 2016-04-12 13162 8.5 8.5 0 1.87999999523163 0.550000011920929 6.05999994277954 0 25 13 328 728 1985 1624580081 2016-04-12 8163 5.30999994277954 5.30999994277954 0 0 0 5.30999994277954 0 0 0 146 1294 1432 1644430081 2016-04-12 10694 7.76999998092651 7.76999998092651 0 0.140000000596046 2.29999995231628 5.32999992370605 0 2 51 256 1131 3199 1844505072 2016-04-12 6697 4.42999982833862 4.42999982833862 0 0 0 4.42999982833862 0 0 0 339 1101 2030 1927972279 2016-04-12 678 0.469999998807907 0.469999998807907 0 0 0 0.469999998807907 0 0 0 55 734 2220

Id SleepDay TotalSleepRecords TotalMinutesAsleep TotalTimeInBed 1503960366 2016-04-12 1 327 346 1927972279 2016-04-12 3 750 775 2026352035 2016-04-12 1 503 546 3977333714 2016-04-12 1 274 469 4020332650 2016-04-12 1 501 541

Id Date WeightKg WeightPounds Fat BMI IsManualReport LogId 6962181067 2016-04-12 62.5 137.788913865625 NULL 24.3899993896484 1 1460505599000 8877689391 2016-04-12 85.8000030517578 189.156627682704 NULL 25.6800003051758 0 1460443631000 1927972279 2016-04-13 133.5 294.317120016975 NULL 47.5400009155273 0 1460509732000 6962181067 2016-04-13 62.0999984741211 136.907061452898 NULL 24.2399997711182 1 1460591999000 8877689391 2016-04-13 84.9000015258789 187.172463959052 NULL 25.4099998474121 0 1460530500000

<span style="color: #008000;">Now that we have an idea of what each of these 3 tables look like, let's take a look at exactly how many participants there are. This</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">will give us an idea of the sample size we're working with. We'll be using the UNION syntax to accomplish this.&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Note to Self: The results from the query below will only show DISTINCT results thus duplicate results will not appear. This is what we</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">want. To include duplicate results when involving UNION please remember to use UNION ALL.</span>

In [1]:
--- Joining all tables to find the total number of participants in the FitBit tracker dataset

SELECT Id
FROM CaseStudy1.dbo.Daily_Activity
UNION
SELECT Id
FROM CaseStudy1.dbo.Daily_Sleep
UNION
SELECT Id
FROM CaseStudy1.dbo.WeightLog

(33 rows affected)

Total execution time: 00:00:00.127

Id 1644430081 7086361926 4702921684 3977333714 6290855005 3372868164 8877689391 2026352035 5577150313 4319703577 4020332650 4057192912 8053475328 4445114986 6962181067 8792009665 2022484408 8378563200 1927972279 2320127002 5553957443 7007744171 1503960366 8583815059 4558609924 2873212765 4388161847 6775888955 1624580081 1844505072 6117666160 2347167796 8253242879

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">We find that there are a total of 33 participants who were involved in our FitBit tracker dataset. This finding is based off&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">of combining the 3 tables (Daily_Activity, Daily_Sleep, and WeightLog) together and looking at the total number of distinct Ids. Moving&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">forward with this analysis, it may be important to note that a sample size of 33 participants may not be sufficient when representing&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">the total population of individuals who use smart devices that track their health.</span>

In [2]:
-- Identifying the number of participants in the Daily_Activity table

SELECT 
    COUNT(DISTINCT Id) AS Number_of_Participants
FROM
    CaseStudy1.dbo.Daily_Activity

(1 row affected)

Total execution time: 00:00:00.027

Number_of_Participants 33

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">There are 33 participants in the Daily_Activity table. It looks like all of the participants in our FitBit tracker dataset&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">happen to have entered an activity log.&nbsp;</span>  

<span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Let's try and find out if these individuals are consistent loggers by looking at how frequently</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">they log their daily activities!</span>

In [3]:
SELECT 
    Id,
    MIN(ActivityDate) AS First_Log,
    MAX(ActivityDate) AS Last_Log,
    COUNT(*) AS Number_Of_Logs
FROM
    CaseStudy1.dbo.Daily_Activity
GROUP BY Id
ORDER BY Number_Of_Logs ASC

(33 rows affected)

Total execution time: 00:00:00.044

Id First_Log Last_Log Number_Of_Logs 4057192912 2016-04-12 2016-04-15 4 2347167796 2016-04-12 2016-04-29 18 8253242879 2016-04-12 2016-04-30 19 3372868164 2016-04-12 2016-05-01 20 6775888955 2016-04-12 2016-05-07 26 7007744171 2016-04-12 2016-05-07 26 6117666160 2016-04-12 2016-05-09 28 6290855005 2016-04-12 2016-05-10 29 8792009665 2016-04-12 2016-05-10 29 3977333714 2016-04-12 2016-05-11 30 1644430081 2016-04-12 2016-05-11 30 5577150313 2016-04-12 2016-05-11 30 4319703577 2016-04-12 2016-05-12 31 4020332650 2016-04-12 2016-05-12 31 8877689391 2016-04-12 2016-05-12 31 2026352035 2016-04-12 2016-05-12 31 7086361926 2016-04-12 2016-05-12 31 4702921684 2016-04-12 2016-05-12 31 2022484408 2016-04-12 2016-05-12 31 8378563200 2016-04-12 2016-05-12 31 1927972279 2016-04-12 2016-05-12 31 2320127002 2016-04-12 2016-05-12 31 5553957443 2016-04-12 2016-05-12 31 8053475328 2016-04-12 2016-05-12 31 4445114986 2016-04-12 2016-05-12 31 6962181067 2016-04-12 2016-05-12 31 1503960366 2016-04-12 2016-05-12 31 8583815059 2016-04-12 2016-05-12 31 4558609924 2016-04-12 2016-05-12 31 2873212765 2016-04-12 2016-05-12 31 4388161847 2016-04-12 2016-05-12 31 1624580081 2016-04-12 2016-05-12 31 1844505072 2016-04-12 2016-05-12 31

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">We find that many of our daily activity loggers were frequent loggers. We only had one outlier which was for participant&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">#4057192019. This individual only had 4 logs, whereas the second most infrequent logger had 18 logs.</span>

<span style="color: #008000;">Next, let's take a look at what our average number of logs were among the 33 participants in the Daily_Activity table.</span>

In [4]:
SELECT AVG(Num_Of_Logs) AS Average_Number_Logs
FROM (SELECT COUNT(*) AS Num_Of_Logs
    FROM CaseStudy1.dbo.Daily_Activity
    GROUP BY Id) AS Table1

(1 row affected)

Total execution time: 00:00:00.018

Average_Number_Logs 28

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">We discover that as an average our 33 participants logged on 28 different days during the time frame we have data for. This show that the&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">participants in our sample size seem to use this activity tracking feature&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">relatively frequently.</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">We'd have to compare this average with the average logging frequencies found in the other 2 datasets, Daily_Sleep and WeightLog tables,</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">to get a better idea of whether an average of 28 logs is considerably high or low.&nbsp;</span>  

<span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">Next, let's take a look at how many participants in our total dataset are in our WeightLog dataset</span>

In [9]:
-- Identifying the number of participants in the WeightLog data table

SELECT 
    COUNT(DISTINCT Id) AS Number_Of_Participants
FROM CaseStudy1.dbo.WeightLog

(1 row affected)

Total execution time: 00:00:00.011

Number_Of_Participants 8

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">We see that 8 participants had used the weightlog function in their FitBit device. This suggests that only 8/33 candidates used </span> <span style="color: rgb(0, 128, 0);">their FitBit devices to track their weights. That's only 24.24% of our sample size!</span>

### <span style="color: #008000;">Hypotheses:</span>

<span style="color: #008000;">From an early perception, these results lead me to believe that perhaps weight logging features should not be a major focus for</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Bellabeats and their product development / marketing campaigns as 75.76% of the smart device users don't use tend to weight log function.&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">But, let's find out a bit more and see exactly how frequently these 8 individuals logged their weights</span>

In [7]:
SELECT 
    Id,
    MIN(Date) AS Earliest_WeightLog,
    MAX(Date) AS Latest_WeightLog,
    COUNT(*) AS Number_of_Logs
FROM 
    CaseStudy1.dbo.WeightLog
GROUP BY Id

(8 rows affected)

Total execution time: 00:00:00.011

Id Earliest_WeightLog Latest_WeightLog Number_of_Logs 1503960366 2016-05-02 2016-05-03 2 1927972279 2016-04-13 2016-04-13 1 2873212765 2016-04-21 2016-05-12 2 4319703577 2016-04-17 2016-05-04 2 4558609924 2016-04-18 2016-05-09 5 5577150313 2016-04-17 2016-04-17 1 6962181067 2016-04-12 2016-05-12 30 8877689391 2016-04-12 2016-05-12 24

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">We see that of our 8 participants who logged their weights in their FitBit device, only 2/8 users were truly serious about </span> <span style="color: rgb(0, 128, 0);">keeping track of their weights on a day to day basis. Users '696121067' and '8877689391' had logged their weights on most if not all of the days that we have their data for</span><span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">.</span>

<span style="color: #008000;">We also noticed that there were 2 users who had logged their weights on only one occasion; users '1927972279' and '55771590313'. These users </span> <span style="color: rgb(0, 128, 0);">may have may have simply lost interest or lacked commitment in weight logging, however, we'd need to have a dataset that tracks them for a</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">longer time horizon to verify if these users simply weightlog less frequently (i.e. once a month) or if they indeed stopped weight </span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">logging altogether.</span>

<span style="color: #008000;">As for the remaining 4/8 users who had logged more than once but also logged substantially less frequently than our daily weight loggers, </span> <span style="color: rgb(0, 128, 0);">3/4 of them logged twice in our dataset. Of these 3 users, we notice that one of them had logged their weights on two consecutive days; this&nbsp;</span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">was user #1503960366. From this we infer that this user had simply given up on weight logging as they had no identifiable patterns. As for </span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">the other two users who had only logged twice, identifying the interval between those two logs would hardly provide enough data to verify if&nbsp;</span>  <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">such an interval would qualify as a pattern due to the lack of ability to verify the consistency of that interval. We'd need a longer time </span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">horizon to accomplish that. In terms of the 4th user who had logged 5 times, let's see if they had any pattern.</span>

In [6]:
--- Taking a closer look at the user who had logged their weight on 5 different occasions

SELECT 
    Id,
    Date
FROM 
    CaseStudy1.dbo.WeightLog
WHERE 
    Id LIKE '4558%'

(5 rows affected)

Total execution time: 00:00:00.007

Id Date 4558609924 2016-04-18 4558609924 2016-04-25 4558609924 2016-05-01 4558609924 2016-05-02 4558609924 2016-05-09

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">After isolating user #4558609924's weight log history, we find that this user had a relatively consistent weight logging pattern. </span> <span style="color: rgb(0, 128, 0);">This user logged at roughly 7 day intervals between their first and second log, second and third, and fourth and fifth. It was only between</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">their third and fourth log had they logged on consecutive days.</span>

<span style="color: #008000;">Based on the data we have the only conclusive thing we could say is that only 3/33 (9.09%) of FitBit users had an interest or could stay </span> <span style="color: rgb(0, 128, 0);">comitted to weight logging on a consistent basis. Consistent meaning that they had either logged at determinable consistent intervals</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">such as user #4558609924 or were like users #696121067 and #8877689391 who had logged almost every single day.</span>

<span style="color: #008000;">Finally, let's take a look at what the average number of logs were in the WeightLog table among the 8 participants</span>

In [5]:
SELECT AVG(Number_of_Logs) AS Average_Number_of_Logs
FROM (
    SELECT 
        Id,
        COUNT(*) AS Number_of_Logs
    FROM
        CaseStudy1.dbo.WeightLog
    GROUP BY Id
) as Table2

(1 row affected)

Total execution time: 00:00:00.018

Average_Number_of_Logs 8

### <span style="color: #008000;">Results:</span>

<span style="color: #008000;">The average number of weight logs submittd by the 8 users who had weight logged was 8. Comparing this average to </span> <span style="color: rgb(0, 128, 0);">the average number of logs for users who had logged their 'daily activities', which was 28, it seems evident that weight logging isn't&nbsp;</span>  <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">a priority amongst the sample population for our FitBit users.</span>

In [10]:
-- Next, let's see if we can find out if our most consistent weight logger ended up losing any weight? 

SELECT 
    Id,
    WeightPounds
FROM CaseStudy1.dbo.WeightLog
WHERE 
    Id = 6962181067 AND 
    Date = (
        SELECT 
            MIN(Date) AS Min
        FROM CaseStudy1.dbo.WeightLog
        WHERE Id = 6962181067 
            ) OR
    Id = 6962181067 AND
    Date = (
        SELECT 
            MAX(Date) AS Max
        FROM CaseStudy1.dbo.WeightLog
        WHERE Id = 6962181067
            )

(2 rows affected)

Total execution time: 00:00:00.032

Id WeightPounds 6962181067 137.788913865625 6962181067 136.466143656502

### <span style="color: #008000;">Note:</span>

<span style="color: #008000;">The WHERE clause in the query above is split up into 2 conditions: (1) Where ID = 6962181067 AND the first day that participant </span> <span style="color: rgb(0, 128, 0);">#6962181067 had logged their weight and (2) where, again, ID = 6962181067 AND the last day that the participant had logged their weight</span>

### <span style="color: #008000;">Results:&nbsp;</span> 

<span style="color: #008000;">Based on the query above we've discovered that from the first day and last day that participant #6962181067 had logged their </span> <span style="color: rgb(0, 128, 0);">weight, this individual had lost roughly 1.3228 pounds (137.7889 lbs - 136.4661 lbs)</span>

### <span style="color: #008000;">Hypothesis:&nbsp;</span> 

<span style="color: #008000;">The motivation behind why #6962181067 was so committed on keeping track of their weight every single day could have </span> <span style="color: rgb(0, 128, 0);">been from their desire to track their weight loss journey!</span>

In [8]:
-- Let's see how user #6962181067's weight fluctuated during the month he had consistently weight logged

SELECT 
    Date,
    WeightPounds AS Weight_lbs
FROM
    CaseStudy1.dbo.WeightLog
WHERE 
    Id = 6962181067
ORDER BY
    Date ASC

(30 rows affected)

Total execution time: 00:00:00.015

Date Weight_lbs 2016-04-12 137.788913865625 2016-04-13 136.907061452898 2016-04-14 136.025217450139 2016-04-15 135.584291243775 2016-04-16 136.6866025547 2016-04-17 135.363832345577 2016-04-18 134.922906139214 2016-04-19 135.363832345577 2016-04-20 136.025217450139 2016-04-21 135.363832345577 2016-04-22 135.363832345577 2016-04-23 135.584291243775 2016-04-24 135.584291243775 2016-04-25 136.025217450139 2016-04-27 134.922906139214 2016-04-28 134.922906139214 2016-04-29 135.363832345577 2016-04-30 134.48197993285 2016-05-01 136.025217450139 2016-05-02 135.584291243775 2016-05-03 134.48197993285 2016-05-04 134.702438831048 2016-05-05 135.143365037411 2016-05-06 135.584291243775 2016-05-07 134.922906139214 2016-05-08 134.922906139214 2016-05-09 137.568454967427 2016-05-10 136.907061452898 2016-05-11 136.466143656502 2016-05-12 136.466143656502

<span style="color: #008000;">It might be a bit difficult identifying the trend of user 6962181067's weight as the month went by purely based on the table below;&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">let's try visualizing by exporting this table and importing it into Tableau!&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Visualized result can be found here:</span> [https://public.tableau.com/views/BellabeatDataProject-User6962181067sWeightLog/Sheet1?:language=en-US&publish=yes&:display\_count=n&:origin=viz\_share\_link](https://public.tableau.com/views/BellabeatDataProject-User6962181067sWeightLog/Sheet1?:language=en-US&publish=yes&:display_count=n&:origin=viz_share_link)

### <span style="color: #008000;">Result:&nbsp;</span> 

<span style="color: #008000;">There we're some fluctuations in our user's weight but generally speaking it looks like our most active weight-logger had a&nbsp;</span>    <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">downward trend in terms of his/her overall weight throughout April 12th to May 12th.&nbsp;</span> <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Discrepancies between weight could be explained by numerous reasons. Here's just to list a few:</span>

1. <span style="color: #008000;">The human body's weight naturally fluctuates, this could be what we're observing from our visual</span>
2. <span style="color: #008000;">The individual may have logged consistently throughout the month; for example before a meal on one day and after a meal on another</span>
3. <span style="color: #008000;">Water intake and expiration based on exercise intensity may also impact weight</span>

<span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Next, let's take a look at how many participants logged their sleep using their FitBit device</span>

In [11]:
-- Now let's take a look at how many participants there were in the Daily_Sleep dataset

SELECT
    COUNT(DISTINCT Id) Number_of_Sleep_Loggers
FROM 
    CaseStudy1.dbo.Daily_Sleep

(1 row affected)

Total execution time: 00:00:00.010

Number_of_Sleep_Loggers 24

### <span style="color: #008000;">Result:</span>

<span style="color: #008000;">Our finding shows that 24/33 users had used their FitBit device to track their sleep. That's roughly 72.72% of our sample </span> <span style="color: rgb(0, 128, 0);">population. Right of the bat, we notice that many more FitBit users are concerned with logging their sleep than they are their weight. </span> <span style="color: rgb(0, 128, 0);">Next, just to satisfy my curiosity, let's check out how many of our weight loggers also logged their sleep:</span>

In [14]:
SELECT 
    COUNT(DISTINCT Id) AS Number_Of_WeightSleep_Loggers
FROM
    CaseStudy1.dbo.Daily_Sleep
WHERE Id IN (
    SELECT 
        DISTINCT Id
    FROM 
        CaseStudy1.dbo.WeightLog
        )

(1 row affected)

Total execution time: 00:00:00.007

Number_Of_WeightSleep_Loggers 6

### <span style="color: rgb(0, 128, 0); font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Result:</span>

<span style="color: #008000;">We find that 6/8 of our weight loggers had also kept a sleep log!&nbsp;</span>  <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">Now let's take a look at whether the individuals who sleep logged were more consistent than the individuals who weight logged when it comes to actively inputting </span> <span style="font-family: -apple-system, BlinkMacSystemFont, sans-serif; color: rgb(0, 128, 0);">data into their FitBit device.</span>

In [1]:
SELECT
    Id,
    MIN(SleepDay) AS Earliest_Sleep_Log,
    MAX(SleepDay) AS Latest_Sleep_log,
    Count(*) AS Number_Of_Sleep_Logs
FROM
    CaseStudy1.dbo.Daily_Sleep
GROUP BY 
    Id
ORDER BY Number_Of_Sleep_logs DESC

(24 rows affected)

Total execution time: 00:00:00.104

Id Earliest_Sleep_Log Latest_Sleep_log Number_Of_Sleep_Logs 8378563200 2016-04-12 2016-05-12 32 6962181067 2016-04-12 2016-05-12 31 5553957443 2016-04-12 2016-05-12 31 4445114986 2016-04-12 2016-05-12 28 2026352035 2016-04-12 2016-05-12 28 3977333714 2016-04-12 2016-05-10 28 4702921684 2016-04-12 2016-05-12 28 4319703577 2016-04-14 2016-05-12 26 5577150313 2016-04-12 2016-05-11 26 1503960366 2016-04-12 2016-05-11 25 4388161847 2016-04-15 2016-05-11 24 7086361926 2016-04-12 2016-05-12 24 6117666160 2016-04-16 2016-05-09 18 2347167796 2016-04-13 2016-04-29 15 8792009665 2016-04-12 2016-05-04 15 4020332650 2016-04-12 2016-05-10 8 1927972279 2016-04-12 2016-04-28 5 4558609924 2016-04-21 2016-05-08 5 1644430081 2016-04-29 2016-05-08 4 1844505072 2016-04-15 2016-05-01 3 6775888955 2016-04-13 2016-04-15 3 8053475328 2016-04-20 2016-05-07 3 7007744171 2016-04-16 2016-05-01 2 2320127002 2016-04-23 2016-04-23 1

### <span style="color: #008000;">Result:</span>

<span style="color: #008000;">We've created a query where we have been able to identify the first and last date that each of </span> <span style="color: rgb(0, 128, 0);">the 24 participants had logged their sleep. We were also able to identify how many times each individual </span> <span style="color: rgb(0, 128, 0);">had logged their sleep between their first and last sleep log date.</span>

  

<span style="color: #008000;">We immediately notice that most participants who logged their sleep were more consistent with logging </span> <span style="color: rgb(0, 128, 0);">than those who had logged their weights! Next, let's see what the average number of sleep logs were </span> <span style="color: rgb(0, 128, 0);">among all the participants!</span>

In [13]:
SELECT 
    MIN(SleepDay) AS Earliest_Log_Date,
    MAX(SleepDay) AS Latest_Log_Date,
    (SELECT AVG(Num_Of_Logs) -- Reminder (for myself): This is a subquery within the SELECT clause
    FROM (SELECT 
            Id,
            COUNT(*) AS Num_Of_Logs
        FROM 
            CaseStudy1.dbo.Daily_Sleep
        GROUP BY Id
        ) AS subquery1
    ) AS Average_Num_of_Logs
FROM 
    CaseStudy1.dbo.Daily_Sleep

(1 row affected)

Total execution time: 00:00:00.016

Earliest_Log_Date Latest_Log_Date Average_Num_of_Logs 2016-04-12 2016-05-12 17

### <span style="color: #008000;">Result:</span>

<span style="color: #008000;">We find that between the first sleep log recorded and the last sleep log recorded for data collected within the Daily_Sleep </span> <span style="color: rgb(0, 128, 0);">table, the average number of logs made amongst the 24 participants was 17!</span>

## Final Remarks:

## <span style="color: rgb(0, 128, 0); font-size: 14px; font-family: -apple-system, BlinkMacSystemFont, sans-serif;">Let's start off by summarizing the findings we've made with our analysis on the FitBit tracker dataset using SQL:</span>  

- <span style="color: #008000;">We had a sample size of 33 FitBit users</span>
- <span style="color: #008000;">All of the users in our dataset (33/33) happened to use their FitBit devices to track their Daily Activities; the average number of logs among all these participants was 28</span>
- <span style="color: #008000;">Only 8/33 or 24.24% of our users had used their FitBit devices for weight logging, of which we could only define 3 of the 33 as consistent weight loggers given the data available. Additionally, the average logs submitted between the users in our weight log dataset was only 8!</span>
- <span style="color: #008000;">Sleep loggers were more plentiful; 24/33 or 72.72% of our sample size had used their FitBit devices to track their sleep. Between our sleep loggers, they averaged 17 logs submitted.</span>

### \*Log Frequency Dashboard - Tableau\*

A dashboard created on Tableau illustrating the number of logs each user made for each health category tracked (Daily\_Activity, Daily\_Sleep, and WeightLog) can be found [here](https://public.tableau.com/app/profile/sunny1222/viz/BellabeatsCaseStudy-UserLoggingFrequencyDashboard/Dashboard1).

<span style="color: #008000;">Based on these findings, the conclusion for Bellabeats is the following:</span>

- <span style="color: #008000;">Asides from developing a product that focuses on tracking one's daily activity, it's also important to focus on marketing and perhaps further improving the sleep tracking feature for Bellabeat devices.</span>
- <span style="color: #008000;">While the weight logging aspect of these heath-tech smart devices might not be a high priority due to its low share of users, discovering a way to make weight logging easier (or perhaps more automated) could increase the number of individuals interested in the weight logging</span>